src/app/clusters/camera-av-stream-management-server/camera-av-stream-management-server.cpp

```diff
@@ -0,0 +1,1801 @@
+    }
+    else
+    {
+        ChipLogDetail(Zcl, "CameraAVStreamMgmt: Unable to load the LocalVideoRecordingEnabled from the KVS. Defaulting to %u",
+                      mLocalVideoRecordingEnabled);
+    }
+
+    // Load LocalSnapshotRecordingEnabled
+    bool localSnapshotRecordingEnabled;
+    err = GetSafeAttributePersistenceProvider()->ReadScalarValue(
+        ConcreteAttributePath(mEndpointId, mClusterId, Attributes::LocalSnapshotRecordingEnabled::Id),
+        localSnapshotRecordingEnabled);
+    if (err == CHIP_NO_ERROR)
+    {
+        mLocalSnapshotRecordingEnabled = localSnapshotRecordingEnabled;
+        ChipLogDetail(Zcl, "CameraAVStreamMgmt: Loaded LocalSnapshotRecordingEnabled as %u", mLocalSnapshotRecordingEnabled);
+    }
+    else
+    {
+        ChipLogDetail(Zcl, "CameraAVStreamMgmt: Unable to load the LocalSnapshotRecordingEnabled from the KVS. Defaulting to %u",
+                      mLocalSnapshotRecordingEnabled);
+    }
+
+    // Load StatusLightEnabled
+    bool statusLightEnabled;
+    err = GetSafeAttributePersistenceProvider()->ReadScalarValue(
+        ConcreteAttributePath(mEndpointId, mClusterId, Attributes::StatusLightEnabled::Id), statusLightEnabled);
+    if (err == CHIP_NO_ERROR)
+    {
+        mStatusLightEnabled = statusLightEnabled;
+        ChipLogDetail(Zcl, "CameraAVStreamMgmt: Loaded StatusLightEnabled as %u", mStatusLightEnabled);
+    }
+    else
+    {
+        ChipLogDetail(Zcl, "CameraAVStreamMgmt: Unable to load the StatusLightEnabled from the KVS. Defaulting to %u",
+                      mStatusLightEnabled);
+    }
+
+    // Load StatusLightBrightness
+    uint8_t statusLightBrightness;
+    err = GetSafeAttributePersistenceProvider()->ReadScalarValue(
+        ConcreteAttributePath(mEndpointId, mClusterId, Attributes::StatusLightBrightness::Id), statusLightBrightness);
+    if (err == CHIP_NO_ERROR)
+    {
+        mStatusLightBrightness = static_cast<Globals::ThreeLevelAutoEnum>(statusLightBrightness);
+        ChipLogDetail(Zcl, "CameraAVStreamMgmt: Loaded StatusLightBrightness as %d", to_underlying(mStatusLightBrightness));
+    }
+    else
+    {
+        ChipLogDetail(Zcl, "CameraAVStreamMgmt: Unable to load the StatusLightBrightness from the KVS. Defaulting to %d",
+                      to_underlying(mStatusLightBrightness));
+    }
+
+    // Signal delegate that all persistent configuration attributes have been loaded.
+    mDelegate.PersistentAttributesLoadedCallback();
+}
+
+CHIP_ERROR CameraAVStreamMgmtServer::StoreViewport(const ViewportStruct & viewport)
+{
+    uint8_t buffer[kViewportStructMaxSerializedSize];
+    TLV::TLVWriter writer;
+
+    writer.Init(buffer);
+    ReturnErrorOnFailure(viewport.Encode(writer, TLV::AnonymousTag()));
+
+    return mPersistentStorage->SyncSetKeyValue(DefaultStorageKeyAllocator::CameraAVStreamMgmtViewport().KeyName(), buffer,
+                                               static_cast<uint16_t>(writer.GetLengthWritten()));
```

Comment:
I do wonder: is there a reason to not use the key format that the other attributes (via SafeAttributePersistenceProvider) use?

And the reason I ask that is this: as soon as you have two instances of this cluster on different endpoints, this setup breaks completely, with the global key name you have here.